src/Hazelcast.Net/Predicates/Predicate.cs

```diff
@@ -12,119 +12,86 @@
-        }
-
-        public static SqlPredicate Sql(string sql)
-        {
-            return new SqlPredicate(sql);
-        }
-
-        public static TruePredicate True()
-        {
-            return new TruePredicate();
-        }
+            => new PredicateProperty(ThisConst);
+
+        public static IPredicate InstanceOf(string fullJavaClassName)
+            => new InstanceofPredicate(fullJavaClassName);
+
+        public static IPredicate And(params IPredicate[] predicates)
+            => new AndPredicate(predicates);
+
+        public static IPredicate False()
+            => new FalsePredicate();
+
+        public static IPredicate IsBetween(string attributeName, object from, object to)
+            => new BetweenPredicate(attributeName, from, to);
+
+        public static IPredicate IsEqual(string attributeName, object value)
+            => new EqualPredicate(attributeName, value);
+
+        public static IPredicate IsGreaterThan(string attributeName, object value)
+            => new GreaterLessPredicate(attributeName, value, false, false);
+
+        public static IPredicate IsGreaterThanOrEqual(string attributeName, object value)
+            => new GreaterLessPredicate(attributeName, value, true, false);
+
+        public static IPredicate IsILike(string attributeName, string expression)
+            => new CaseInsensitiveLikePredicate(attributeName, expression);
+
+        public static IPredicate IsIn(string attributeName, params object[] values)
+            => new InPredicate(attributeName, values);
+
+        public static IPredicate IsLessThan(string attributeName, object value)
+            => new GreaterLessPredicate(attributeName, value, false, true);
+
+        public static IPredicate IsLessThanOrEqual(string attributeName, object value)
+            => new GreaterLessPredicate(attributeName, value, true, true);
+
+        public static IPredicate IsLike(string attributeName, string expression)
+            => new LikePredicate(attributeName, expression);
+
+        public static IPredicate IsNotEqual(string attributeName, object value)
+            => new NotEqualPredicate(attributeName, value);
+
+        public static IPredicate MatchesRegex(string attributeName, string regex)
+            => new RegexPredicate(attributeName, regex);
+
+        public static IPredicate Not(IPredicate predicate)
+        => new NotPredicate(predicate);
+
+        public static IPredicate Or(params IPredicate[] predicates)
+            => new OrPredicate(predicates);
+
+        public static IPredicate Sql(string sql)
+            => new SqlPredicate(sql);
+
+        public static IPredicate True()
+            => new TruePredicate();
+
+        public static IPredicate Partition(object partitionKey, IPredicate predicate)
+            => new PartitionPredicate(partitionKey, predicate);
+
+        public static IPredicate Page(int pageSize, IPredicate predicate, IComparer<KeyValuePair<object, object>> comparer)
```

Comment:
This should return something like `IPagingPredicate`. It has some extra methods defined. See https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/query/PagingPredicate.java

Reply:
I can see it in Java yes, now I am wondering whether all properties need to be exposed in `IPagingPredicate`? Java would have `NextPage`, `PreviousPage`, `Page`, `PageSize`, `Anchor` and a `Reset()` method, are they all useful for users?

Meh - I also note that in Java these two predicates are typed ie it's `PartitionPredicate<K,V>` but totally not the case in C#... need to look into it...

Reply:
Right, I have pushed a fix to the PR that implements the two interfaces.

I still have concerns about the partition predicate being typed (`<K,V>`) in Java vs being `<object,object>` in C# but I have put a `// FIXME` comment in the code, so if it looks good enough for you I propose we merge this PR, and fix later.